src/modules/organizations/containers/edit-details-container.jsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import apiClient from 'panoptes-client/lib/api-client';
+import EditDetails from '../components/edit-details';
+import { organizationShape, organizationAvatarShape, organizationBackgroundShape } from '../model';
+import { setCurrentOrganization, setOrganizationAvatar, setOrganizationBackground } from '../action-creators';
+
+class EditDetailsContainer extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.fetchAvatar = this.fetchAvatar.bind(this);
+    this.fetchBackground = this.fetchBackground.bind(this);
+    this.handleMediaChange = this.handleMediaChange.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.props.organization) {
+      this.fetchAvatar(this.props.organization);
+      this.fetchBackground(this.props.organization);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.organization !== nextProps.organization) {
+      this.fetchAvatar(nextProps.organization);
+      this.fetchBackground(nextProps.organization);
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.dispatch(setOrganizationAvatar({}));
+    this.props.dispatch(setOrganizationBackground({}));
+  }
+
+  fetchAvatar(org) {
+    if (org.links.avatar.id) {
+      apiClient.type('avatars').get(org.links.avatar.id)
+        .then((avatar) => {
+          this.props.dispatch(setOrganizationAvatar(avatar));
+        }).catch((error) => {
+          if (error.status !== 404) console.error(error);
+        });
+    }
+  }
+
+  fetchBackground(org) {
+    if (org.links.background.id) {
+      apiClient.type('backgrounds').get(org.links.background.id)
+        .then((background) => {
+          this.props.dispatch(setOrganizationBackground(background));
+        }).catch((error) => {
+          if (error.status !== 404) console.error(error);
+        });
+    }
+  }
+
+  handleMediaChange(type, file) {
```

Comment:
Yeah, I think there's a bug with the cache since I'm trying to use `refresh` (see: https://github.com/zooniverse/json-api-client/blob/master/src/resource.coffee#L76-L84). I think this is why there are 404s and inconsistent re-renders. I'll address in a follow up PR.